### ISSUE TITLE
Non deterministic soft navigation detection after hard navigation

### ISSUE THREAD

**Issue by ErikWitt** [WICG/soft-navigations#41]
Testing the soft navigation via origin trial on this site (https://www.tom-tailor.de/), I am experiencing that after a hard navigation or reload, the first few soft navigations are not detected (even though the origin trial token is already there).

It varies how many soft navigations are not detected and after the first one is detected, it seems to work consistently. However it is very easy to reproduce that at least the first navigation is not captured.

The way I reproduce this is
1. Hard navigation or reload to https://www.tom-tailor.de/damen-startseite
2. Navigate to /sale by clicking the red link in the header
3. Navigate back by clicking the icon in the middle of the header
4. Repeat

Expected: every navigation should be captured as a soft navigation.
Actual: the first navigations are not captured. At some point navigations are captured consistently even though it is the same navigation every time.

I tested on Chrome Version 120.0.6099.129 on Mac OS 14.1.1 (23B81). I also wanted to attach a video showcasing this but screen captures are huge, so here is a drive link: https://drive.google.com/file/d/1lvFLDYmE_6gVOVQJaoceRRa8SIFy45A9/view?usp=sharing

**Comment by clelland** [comment]
Hi @ErikWitt -- could you for a bug for this at https://crbug.com/ ? It sounds like it is a Chrome implementation issue (at least, it's not obvious to me that it's a problem with the definitions here, and it sounds more like Chrome is misbehaving)

**Comment by ErikWitt** [comment]
Agreed, I have filed a bug report: https://bugs.chromium.org/p/chromium/issues/detail?id=1515268
